src/Xamarin.Android.Build.Tasks/Xamarin.Android.Common.targets

```diff
@@ -2866,6 +2896,56 @@ because xbuild doesn't support framework reference assemblies.
 	</ItemGroup>
 </Target>
 
+<Target Name="_PrepareNativeAssemblyItems">
+  <ItemGroup>
+    <_NativeAssemblyTarget Include="@(_TypeMapAssemblySource->Replace('.s', '.o'))">
+      <abi>%(_TypeMapAssemblySource.abi)</abi>
+    </_NativeAssemblyTarget>
+    <_NativeAssemblyTarget Include="@(_EnvironmentAssemblySource->Replace('.s', '.o'))">
+      <abi>%(_EnvironmentAssemblySource.abi)</abi>
+    </_NativeAssemblyTarget>
+  </ItemGroup>
+</Target>
+
+<Target Name="_CompileNativeAssemblySources"
+        DependsOnTargets="_PrepareNativeAssemblyItems"
+        Inputs="@(_TypeMapAssemblySource);@(_EnvironmentAssemblySource)"
+        Outputs="@(_NativeAssemblyTarget)">
+        <CompileNativeAssembly
+            Sources="@(_TypeMapAssemblySource);@(_EnvironmentAssemblySource)"
+            DebugBuild="$(AndroidIncludeDebugSymbols)"
+            WorkingDirectory="$(_NativeAssemblySourceDir)"/>
+</Target>
```

Comment:
The big question here, is how long this `<CompileNativeAssembly/>` task takes?

Can you attach a binlog of the XF test project in this repo? or the output of `xabuild /clp:performancesummary`?

Reply:
```
        58 ms  LinkApplicationSharedLibraries             1 calls
      242 ms  CompileNativeAssembly                      1 calls
```

This is the time it takes to build around 15MB of .s sources to produce 4 2.6mb shared libraries.